base/src/org/adempiere/process/YearEndClosing.java

```diff
@@ -0,0 +1,186 @@
+				whereclause, get_TrxName())
+		.setClient_ID()
+		.setParameters(elementId)
+		.setOrderBy("value")
+		.list();		
+		for (MElementValue el : accounts) 
+		{			
+			balance = SearchBalance(el.getC_ElementValue_ID());
+			if (balance.equals(Env.ZERO))
+				continue;
+			CreateJournal(el.getC_ElementValue_ID(), journalBatch, acctSchema, balance);
+		}
+		return "@Created@: " + journalBatch.getDocumentNo();		
+	}	//	doIt
+	
+	private int CreateAccount(int elementValueId)
+	{   
+		
+		
+		MAccount acc = MAccount.get(getCtx(), getAD_Client_ID(),
+		getOrgId(), getAcctSchemaId(), 
+		elementValueId, 0,
+		0, 0, 0, 
+		0, 0, 0, 
+		0, 0, 0,
+		0, 0, 0,0,0,0, get_TrxName());
+		if (acc == null)
+			return 0;
+		return acc.get_ID();
+		
+	}
+	
+	private Boolean CreateJournal(int elementValueID, MJournalBatch journalBatch, 
+			MAcctSchema acctSchema, BigDecimal balance)
+	{
+		MJournal journal = new MJournal(journalBatch);
+		int accountId = CreateAccount(elementValueID);
+		MElementValue elementValue = new MElementValue(getCtx(), elementValueID, get_TrxName() );
+		
+		journal.setC_Currency_ID(acctSchema.getC_Currency_ID());
+		journal.setC_AcctSchema_ID(acctSchema.get_ID());
+		journal.setC_ConversionType_ID(MConversionType.getDefault(getAD_Client_ID()));
+		journal.setDescription(Msg.translate(Env.getCtx(), "Closing") + " " + elementValue.getValue() + " " + elementValue.getName());
+		journal.setDocumentNo(DB.getDocumentNo(getAD_Client_ID(), MJournal.Table_Name, get_TrxName()));
+		journal.setGL_Category_ID(getCategoryId());
+		journal.save();
+		if (!CreateLines(accountId, journal, elementValueID, balance))
+			return false;				
+		return true;
+	}
+	
+	private Boolean CreateLines(int accountID, MJournal journal, 
+			int elementValueId, BigDecimal balance)
+	{
+		MElementValue el = new MElementValue(getCtx(), elementValueId, get_TrxName());
+		String accounttype = el.getAccountType();
+		MJournalLine dr = new MJournalLine(journal);
+		MJournalLine cr = new MJournalLine(journal);
+		if (balance.signum() == 0)
+			return true;
+		if (accounttype.equals('R'))
+		{
+			dr.setC_ValidCombination_ID(closingAccountId);
+			dr.setAmtSourceDr(balance.negate());
+			dr.save();
+			cr.setC_ValidCombination_ID(accountID);
+			cr.setAmtSourceCr(balance.negate());
+			cr.save();
+		}
+		else
+		{
+			dr.setC_ValidCombination_ID(closingAccountId);
+			dr.setAmtSourceDr(balance);
+			dr.save();
+			cr.setC_ValidCombination_ID(accountID);
+			cr.setAmtSourceCr(balance);
+			cr.save();
+		}
+		return true;
+	}
+	
+	private BigDecimal SearchBalance(int elementValueId)
+	{
+		String sql = " SELECT COALESCE(sum(amtacctDr - amtacctCr), 0) FROM fact_acct f" +
+			" WHERE dateacct BETWEEN Firstof( " + DB.TO_DATE(getDateAcct()) +
+			", 'YY' ) AND " + DB.TO_DATE(getDateAcct()) + 
+			" AND f.ad_client_ID = " + Env.getAD_Client_ID(getCtx()) + 
+			" AND f.account_ID = " + elementValueId;
+		BigDecimal saldo = DB.getSQLValueBD(get_TrxName(), sql);

```

Comment:
@SusanneCalderon Hi Susanne, @marcalwestf Mario, Can you use parameters vs build SQL using string to avoid SQL injection.

We are discouraging the use of this incorrect practice in the code

``` DB.getSQLValueBD(get_TrxName(), sql, parameteres);```